app/routes/goal_routes.py

```diff
@@ -1 +1,75 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, request, Response
+from app.models.goal import Goal
+from app.models.task import Task
+from ..db import db
+from .helper_methods import validate_model, helper_model_from_dict, helper_get_sorted_query
+
+bp = Blueprint("goals_bp", __name__, url_prefix="/goals")
+
+
+
+@bp.post("")
+def create_goal():
+    request_body = request.get_json()
+    return helper_model_from_dict(Goal, request_body, required_fields=["title"])
+
+@bp.post("/<goal_id>/tasks")
+def add_tasks_to_goal_id(goal_id):
+    goal = validate_model(Goal, goal_id)
+    request_body = request.get_json()
+    
+    task_ids = request_body.get("task_ids", [])
+    # goal.tasks.clear()
+    for task in goal.tasks:
+        task.goal_id = None
+    
+    for id in task_ids:
+        task = validate_model(Task, id)
+        task.goal_id = goal.id
+        # goal.tasks.append(task)
+
+    db.session.commit()
+
+    return {
+        "id": int(goal.id),
+        "task_ids": task_ids
+    }, 200
+
+
+@bp.get("")
+def get_goals():
+    sort_param = request.args.get("sort")
+    query = helper_get_sorted_query(Goal, sort_param)
+    goals = db.session.execute(query).scalars().all()
+    return [goal.to_dict() for goal in goals]
+
+@bp.get("/<goal_id>")
+def get_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+    return {"goal": goal.to_dict()}
+
+@bp.get("/<goal_id>/tasks")
+def get_tasks_of_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+    tasks = [task.to_dict() for task in goal.tasks]
+
+    return {
+        "id": goal.id,
+        "title": goal.title,
+        "tasks": tasks
+    }, 200
```

Comment:
Could we use the `to_dict` method in `Goal` instead of repeating this logic?

I'd prefer that you use the method you have written because it makes the route more concise by reducing repetition. You can update the dictionary here in the route to associate tasks with the goal, see suggestion:

```suggestion
    goal = validate_model(Goal, goal_id)
    tasks = [task.to_dict() for task in goal.tasks]

    response = goal.to_dict()
    response["tasks] = tasks
    return response
```